### PR TITLE
Support dynamic batch dimension in LightningModelExporter

### LINKED ISSUE #7016: Exported OpenVINO IR has a static batch dimension (batch=1), breaking inference with batch size > 1
**Summary**
Models exported from Geti (e.g. YOLOX-Tiny) produce an OpenVINO IR with a hard-coded static input shape [1, 3, H, W]. Any consumer that requests batch size > 1 (e.g. DL Streamer Pipeline Server with batch-size=8) fails, because a static batch of 1 cannot be reshaped to 8. The failure is device-independent (GPU/CPU/NPU) since it is a property of the model graph, not the runtime. Both exported and NNCF-optimized IRs are affected identically.

**Root cause**
LightningModelExporter.to_openvino() passes a fully static PartialShape to openvino.convert_model(..., input=...):

native.py:74-96 — input=(openvino.PartialShape(input_size),)
input_size = self.data_input_params.as_ncwh() defaults to batch_size=1 (base.py:105)
The dynamic-batch axis declared for YOLOX ("dynamic_shapes": {"inputs": {0: Dim("batch")}}) is never used because the exporter is constructed with via_onnx=False, so onnx_export_configuration is ignored on the direct Torch→OV path (yolox.py:162-169).

**Steps to reproduce**

Train a YOLOX-Tiny model in Geti 3.0.
Export to OpenVINO IR (and optionally optimize with NNCF).
Inspect the IR:

Run ./sample_start.sh -p pallet_defect_detection_gpu (uses batch-size=8) → dlstreamer-pipeline-server errors out.
Expected
Input partial shape should be [?,3,H,W] (dynamic batch) or at least match the deployment batch size, and benchmark_app -m exported_model.xml -shape "[8,3,H,W]" should succeed.

**Actual**
Input shape is static [1,3,H,W]; batched inference (batch > 1) fails.

**Proposed fix**
Declare a dynamic batch dimension in the OV conversion (PartialShape([-1, C, H, W])), keeping the tracing example_input at batch 1; expose an optional dynamic_batch/batch_size export flag. Verify the YOLOX detection head (NMS/TopK/reshape in export_by_feat) exports cleanly with a dynamic batch; if not, fall back to a fixed batch_size=N export.

**Affected components / scope**
getitune library export path (Lightning backend). Likely affects all Lightning-backend models exported to OpenVINO/ONNX, not only YOLOX.

### PR DESCRIPTION
**Pull request by itallix**
### Summary

- [x] Enabled dynamic dimension for `openvino.PartialShape`  in `LightningModelExporter#to_openvino` path

### Checklist

- [x] The PR title and description are clear and descriptive
- [x] I have manually tested the changes
- [x] All changes are covered by automated tests
- [x] All related issues are linked to this PR (if applicable)
- [ ] Documentation has been updated (if applicable)

### CODE REVIEW

File: library/tests/unit/backend/lightning/exporter/test_native.py [hunk 1]
```diff
@@ -78,3 +92,85 @@ def test_to_onnx_export(self, exporter, dummy_model, tmp_path):
...
+        assert result.shape[0] == self.BATCH_SIZE
+
+    def test_to_onnx_export_dynamic_batch(self, fxt_exporter, fxt_dummy_model, fxt_dummy_input, tmp_path):
+        """Exported ONNX model's input/output batch axis should be dynamic (symbolic), not fixed to 1."""
+        output_dir = tmp_path / "onnx_export_dynamic"
+        output_dir.mkdir()
+
+        fxt_exporter.onnx_export_configuration = {
+            "dynamic_shapes": ({0: torch.export.Dim("batch")},),
+        }
```
[the remark this answers is not in the document]
Comment (itallix) [reply library/tests/unit/backend/lightning/exporter/test_native.py:155]: @copilot positional arg preferred over name lookup to avoid coupling to dummy model structure

File: library/src/getitune/backend/lightning/exporter/base.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
     from getitune.backend.lightning.models.base import DataInputParams, LightningModel
 
 
-class ModelExporter:
+class ModelExporter(ABC):
```
Comment (kprokofi) [inline library/src/getitune/backend/lightning/exporter/base.py:26]: Why should we make it abstract?

File: library/.gitignore [hunk 3]
```diff
@@ -23,6 +23,7 @@ __pycache__/
 benchmark*.csv
 benchmark*.xlsx
 *.raw.csv
+csv/
```
Comment (kprokofi) [inline library/.gitignore]: Why this change?

File: library/src/getitune/backend/lightning/exporter/base.py (hunk 2 above)
Comment (itallix) [reply library/src/getitune/backend/lightning/exporter/base.py:26]: `@abstractmethod` decorator is applied to `to_openvino` and `to_onnx` methods, but it won't enforce any checks by it's own, so instantiating the class succeeds without throwing any error at runtime. To prevent instantiation of un-implemented methods, `ABC` must be added as a base class.

File: library/.gitignore (hunk 3 above)
Comment (itallix) [reply library/.gitignore]: Running `uv run pytest tests/integration/api/test_xai.py -k test_forward_explain` generates a `csv/` directory containing multiple `hparams.yaml` files. Added this line to `.gitignore` so these files aren't tracked by default.

Comment (itallix) [reply library/.gitignore]: I've reverted this change in favor of passing `tmp_path` as `work_dir` to `LightningEngine`.

**Review by kprokofi — APPROVED** [review]

**Review by leoll2 — APPROVED** [review]
